networkx/algorithms/bipartite/tests/test_matching.py

```diff
@@ -82,3 +84,13 @@ def test_to_vertex_cover(self):
         matching = maximum_matching(self.graph)
         vertex_cover = to_vertex_cover(self.graph, matching)
         self.check_vertex_cover(vertex_cover)
+
+
+def test_eppstein_matching():
+    # Special test
+    G = nx.Graph()
+    G.add_nodes_from(['a', 2, 3, 4], bipartite=0)
+    G.add_nodes_from([1, 'b', 'c'], bipartite=1)
+    G.add_edges_from([('a', 1), ('a', 'b'), (2, 'b'),
+                      (2, 'c'), (3, 'c'), (4, 1)])
+    assert_true(nx.is_max_cardinality_matching(G, eppstein_matching(G)))
```

Comment:
I believe you gave an example of a matching that caused problems in the original issue report?